cmd/infra/aws/ec2.go

```diff
@@ -0,0 +1,538 @@
+		})
+		if err != nil {
+			return "", fmt.Errorf("cannot associate private route table with subnet: %w", err)
+		}
+	}
+	return tableID, nil
+}
+
+func (o *CreateInfraOptions) createRouteTable(client ec2iface.EC2API, vpcID, name string) (*ec2.RouteTable, error) {
+	result, err := client.CreateRouteTable(&ec2.CreateRouteTableInput{
+		VpcId:             aws.String(vpcID),
+		TagSpecifications: o.ec2TagSpecifications("route-table", name),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("cannot create route table: %w", err)
+	}
+	return result.RouteTable, nil
+}
+
+func (o *CreateInfraOptions) existingRouteTable(client ec2iface.EC2API, name string) (*ec2.RouteTable, error) {
+	result, err := client.DescribeRouteTables(&ec2.DescribeRouteTablesInput{Filters: o.ec2Filters(name)})
+	if err != nil {
+		return nil, fmt.Errorf("cannot list route tables: %w", err)
+	}
+	if len(result.RouteTables) > 0 {
+		return result.RouteTables[0], nil
+	}
+	return nil, nil
+}
+
+func (o *CreateInfraOptions) hasNATGatewayRoute(table *ec2.RouteTable, natGatewayID string) bool {
+	for _, route := range table.Routes {
+		if aws.StringValue(route.NatGatewayId) == natGatewayID &&
+			aws.StringValue(route.DestinationCidrBlock) == "0.0.0.0/0" {
+			return true
+		}
+	}
+	return false
+}
+
+func (o *CreateInfraOptions) hasInternetGatewayRoute(table *ec2.RouteTable, igwID string) bool {
+	for _, route := range table.Routes {
+		if aws.StringValue(route.GatewayId) == igwID &&
+			aws.StringValue(route.DestinationCidrBlock) == "0.0.0.0/0" {
+			return true
+		}
+	}
+	return false
+}
+
+func (o *CreateInfraOptions) hasAssociatedSubnet(table *ec2.RouteTable, subnetID string) bool {
+	for _, assoc := range table.Associations {
+		if aws.StringValue(assoc.RouteTableId) == subnetID {
+			return true
+		}
+	}
+	return false
+}
+
+func (o *CreateInfraOptions) ec2TagSpecifications(resourceType, name string) []*ec2.TagSpecification {
+	return []*ec2.TagSpecification{
+		{
+			ResourceType: aws.String(resourceType),
+			Tags:         ec2Tags(o.InfraID, name),
+		},
+	}
+}
+
+func (o *CreateInfraOptions) ec2Filters(name string) []*ec2.Filter {
+	filters := []*ec2.Filter{
+		{
+			Name:   aws.String(fmt.Sprintf("tag:%s", clusterTag(o.InfraID))),
+			Values: []*string{aws.String(clusterTagValue)},
+		},
+	}
+	if len(name) > 0 {
+		filters = append(filters, &ec2.Filter{
+			Name:   aws.String("tag:Name"),
+			Values: []*string{aws.String(name)},
+		})
+	}
+	return filters
+}
+
+func clusterTag(infraID string) string {
+	return fmt.Sprintf("kubernetes.io/cluster/%s", infraID)
+}
+
+func ec2Tags(infraID, name string) []*ec2.Tag {
+	tags := []*ec2.Tag{
+		{
+			Key:   aws.String(clusterTag(infraID)),
+			Value: aws.String(clusterTagValue),
+		},
+	}
+	if len(name) > 0 {
+		tags = append(tags, &ec2.Tag{
+			Key:   aws.String("Name"),
+			Value: aws.String(name),
+		})
+	}
+	return tags
+
+}
+
+func waitForCondition(checkFunc func() (bool, error), interval, timeOut time.Duration) error {
```

Comment:
How is this different from `wait.Poll()`?

Reply:
hmm, it's not, I will call that instead, thx

Reply:
fixed